### PR TITLE
TCGA/GEO data set integration init

### PR DESCRIPTION
**Pull request by thomasveith**

### CODE REVIEW

File: R/Find_MisSegregationRate_Surrogate.R [hunk 1]
```diff
@@ -62,24 +63,70 @@ allmat = list()
...
+# tmp=as.matrix(assays(X)[["HNSC_RNASeq2GeneNorm-20160128"]])
+# sampleType=sapply(colnames(tmp), function(x) strsplit(x, "-")[[1]][4])
+# tmp=tmp[,grep("01",sampleType)]
+
+###############################################################################
+### Integrate TCGA & GEO data adjusting for Batch Effects using comBat-seq ####
+###############################################################################
+combined<-data.frame(ncol=1)
+for (name in names(allmat)){
+  combined=cbind(combined, allmat[[name]])
```
Comment (noemiandor) [inline R/Find_MisSegregationRate_Surrogate.R:88]: would make sure you use only genes present in all datasets (via something like `plyr::count(unlist(lapply(allmat,rownames)))`)

File: R/Find_MisSegregationRate_Surrogate.R [hunk 2]
```diff
@@ -62,24 +63,70 @@ allmat = list()
...
+}
+combined = cbind(combined, Z[rownames(allmat$HNSC),])
```
Comment (noemiandor) [inline R/Find_MisSegregationRate_Surrogate.R:90]: same as previous comment

File: R/Find_MisSegregationRate_Surrogate.R [hunk 3]
```diff
@@ -62,24 +63,70 @@ allmat = list()
...
+combined <- combined[,-c(1)]
+combined =as.matrix(combined)
+batch <- c(rep(1, 20), rep(2, 20),rep(3, 20),rep(4, 20),rep(5, 20),rep(6, 20),rep(7, 20),rep(8, 20), rep(9, 20), rep(10, 39))
```
Comment (noemiandor) [inline R/Find_MisSegregationRate_Surrogate.R:93]: `sapply(allmat,ncol)` should give u the second argument of `rep` here without hard coding

File: R/Find_MisSegregationRate_Surrogate.R [hunk 4]
```diff
@@ -62,24 +63,70 @@ allmat = list()
...
+adjusted <- ComBat_seq(combined, batch=batch, group=NULL)
```
Comment (noemiandor) [inline R/Find_MisSegregationRate_Surrogate.R:94]: does `ComBat_seq` (or  the package) have any statistics of the adjustments or visualization plots showing how the numbers have changed? would also add a documentation line above describing what it does

File: R/Find_MisSegregationRate_Surrogate.R [hunk 5]
```diff
@@ -62,24 +63,70 @@ allmat = list()
...
+  pLadjusted[[bat]]<-pLadjusted[[bat]] + 1 ### noise for log scale in boxplot
+}
+boxplot(pLadjusted, log="y")
+title("Adjusted")
+
+
+### add together pq objects for plotting in one boxplot
+plotList<-list()
+for (bat in unique(batch)){
+  plotList[[bat]]<-as.numeric(pq[,bat])
```
Comment (noemiandor) [inline R/Find_MisSegregationRate_Surrogate.R:121]: this assumes `pq` has non-unique column names. Does it?

**Review by noemiandor — CHANGES_REQUESTED** [review]

File: R/Find_MisSegregationRate_Surrogate.R (hunk 5 above)
Comment (thomasveith) [reply R/Find_MisSegregationRate_Surrogate.R:121]: Yes, on line 96 we rename the `pq` object with the `batch` object, thus `pq` has 219 columns, but only 10 unique column names (i.e., there are 10 unique batches).

File: R/Find_MisSegregationRate_Surrogate.R (hunk 4 above)
Comment (thomasveith) [reply R/Find_MisSegregationRate_Surrogate.R:94]: Not really, they suggest doing boxplots to compare output. And of course, in their examples, it is much more of a stark contrast than what we've seen. They also suggest doing PCA to see if there is a strong batch effect in the first place (does the data cluster by batch?). Looking into it.

Comment (thomasveith) [reply R/Find_MisSegregationRate_Surrogate.R:94]: [tsne_10batches.pdf](https://github.com/MathOnco/ploidyEvolution/files/7041617/tsne_10batches.pdf)

And yes I think we need to consider all our TCGA data as a single batch, and GEO as the second one (as opposed to all 10 samples being separate batches).

File: R/Find_MisSegregationRate_Surrogate.R (hunk 1 above)
Comment (thomasveith) [reply R/Find_MisSegregationRate_Surrogate.R]: Ah, that makes sense. Updated in f5caf2b.

File: R/Find_MisSegregationRate_Surrogate.R (hunk 2 above)
Comment (thomasveith) [reply R/Find_MisSegregationRate_Surrogate.R]: Also updated in f5caf2b.

File: R/Find_MisSegregationRate_Surrogate.R (hunk 3 above)
Comment (thomasveith) [reply R/Find_MisSegregationRate_Surrogate.R]: Got it in f5caf2b

File: R/Find_MisSegregationRate_Surrogate.R (hunk 4 above)
Comment (thomasveith) [reply R/Find_MisSegregationRate_Surrogate.R:103]: Actually went back to keeping 10 batches, using only two batches did not change results

**Pull request merged**